anchore_engine/util/docker.py

```diff
@@ -153,6 +155,110 @@ def parse_dockerimage_string(instr, strict=True):
     return ret
 
 
+class DockerImageTag:
```

Comment:
Since this supports tag and digest as the last component, I think a more descriptive name would be "DockerPullReference" so that it's clear this supports the various ways docker can pull the image as opposed to a raw digest or ImageID (which is handled by the DockerImageReference object).